### PR TITLE
Relative SQLite URIs with diff

### PR DESCRIPTION
**Pull request by pimeys**
Closes: https://github.com/prisma/prisma/issues/12265

### CODE REVIEW

File: migration-engine/core/src/state.rs [hunk 1]
```diff
@@ -56,14 +56,16 @@ impl EngineState {
         path: &str,
         f: ConnectorRequest<O>,
     ) -> CoreResult<O> {
+        let config_dir = std::path::Path::new(path).parent();
```
Comment (tomhoule) [inline migration-engine/core/src/state.rs:59]: would that work for relative paths? (e.g. `./schema.prisma`)

File: migration-engine/core/src/state.rs [hunk 2]
```diff
@@ -56,14 +56,16 @@ impl EngineState {
...
         let schema = std::fs::read_to_string(path)
             .map_err(|err| ConnectorError::from_source(err, "Falied to read Prisma schema."))?;
-        self.with_connector_for_schema(&schema, f).await
+        self.with_connector_for_schema(&schema, config_dir, f).await
     }
 
     async fn with_connector_for_schema<O: Send + 'static>(
         &self,
         schema: &str,
+        config_dir: Option<&Path>,
```
Comment (tomhoule) [inline migration-engine/core/src/state.rs:68]: The optionality is problematic, but we'll have to change the JSON-RPC API for some methods to fix that. Could we persist that somewhere?

**Review by tomhoule — APPROVED** [review]
It sucks that we have to do it, but lgtm

File: migration-engine/core/src/state.rs (hunk 1 above)
Comment (pimeys) [reply migration-engine/core/src/state.rs:59]: This is basically just copied from your `db execute` PR :)

Comment (tomhoule) [reply migration-engine/core/src/state.rs:59]: hmm we should check that too

**Pull request merged**